relayer/main/main.go

```diff
@@ -313,9 +321,30 @@ func main() {
 			)
 		})
 	}
+
+	// Handle os signal
+	errGroup.Go(func() error {
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
+		sig := <-sigChan
+		logger.Info("Receive os signal", zap.String("signal", sig.String()))
+
+		// Cancel the parent context
+		// This will cascade to errgroup context
+		cancel()
+
+		// No error for graceful shutdown
+		return nil
+	})
+
 	logger.Info("Initialization complete")
-	err = errGroup.Wait()
-	logger.Error("Relayer exiting.", zap.Error(err))
+	if err := errGroup.Wait(); err != nil {
+		logger.Fatal("Relayer exiting with error.", zap.Error(err))
+		os.Exit(1)
```

Comment:
Generally LGTM but do we need these os.Exit(1)? Both here and sig-agg It seems like the only benefit is to return a non-zero exit code but downside is that we need to ensure that all defers have been executed already by the time we call this.